docs/GUIDES.md

```diff
@@ -177,8 +177,242 @@ Our playing field is a grid of tiles, addressable like a coordinate system:
+  - we can find out about the size in pixels of our window and from that calculate how many cells we can fit
+- set up grid system:
+
+  ```elixir
+  @tile_size 32
+
+  def init(arg, opts) do
+    viewport = opts[:viewport]
+
+    {:ok, %ViewPort.Status{size: {vp_width, vp_height}}} = ViewPort.info(viewport)
+
+    # dimensions of the grid (21x18 tiles, 0-indexed)
+    num_tiles_width = trunc(vp_width / @tile_size)
+    num_tiles_height = trunc(vp_height / @tile_size)
+
+    # the entire game state will be held here
+    state = %{
+      viewport: viewport,
+      width: num_tiles_width,
+      height: num_tiles_height
+    }
+  end
+  ```
+
+- important concept in Scenic: "graph"
+  - graph like DOM: hierarchical set of data that describes how things are drawn onto the scene
+  - graph is immutable and can only be "transformed" via functions
+- set up initial graph at compile time since we're updating all the time anyways:
+
+  ```elixir
+  alias Scenic.ViewPort
+  alias Scenic.Graph
+
+  @graph Graph.build()
+  @tile_size 32
+
+  # initialize the game scene
+  def init(arg, opts) do
+    viewport = opts[:viewport]
+
+    {:ok, %ViewPort.Status{size: {vp_width, vp_height}}} = ViewPort.info(viewport)
+
+    # dimensions of the grid (21x18 tiles, 0-indexed)
+    num_tiles_width = trunc(vp_width / @tile_size)
+    num_tiles_height = trunc(vp_height / @tile_size)
+
+    # the entire game state will be held here
+    state = %{
+      viewport: viewport,
+      width: num_tiles_width,
+      height: num_tiles_height
+    }
+  end
+  ```
+
+- return `{:ok, state}` from `init/2`:
+  
+  ```elixir
+  def init(arg, opts) do
+    viewport = opts[:viewport]
+
+    {:ok, %ViewPort.Status{size: {vp_width, vp_height}}} = ViewPort.info(viewport)
+
+    # dimensions of the grid (21x18 tiles, 0-indexed)
+    num_tiles_width = trunc(vp_width / @tile_size)
+    num_tiles_height = trunc(vp_height / @tile_size)
+
+    # the entire game state will be held here
+    state = %{
+      viewport: viewport,
+      width: num_tiles_width,
+      height: num_tiles_height
+    }
+
+    {:ok, state}
+  end
+  ```
+- optionally give it a more Nokia style feel:
+  - default background for the scene is black, but we can set the background like so:
+
+    ```elixir
+    @graph Graph.build(clear_color: :dark_sea_green)
+    ```
+
+- we need to add our graph to the scene to actually draw something
+- add information about game objects: how can we define a snake?
+  - we describe a snake via:
+  - body: a list of ordered x and y pairs. Each pair corresponds to a cell in the grid the snake is currently occupying
+  - size: the current size of the snake in cells
+
+  ```elixir
+  snake = %{body: [{9, 9}], size: 1}
+  ```
+
+- need to update the graph to actually draw something.
+
+  ```elixir
+  # the entire game state will be held here
+  state = %{
+    viewport: viewport,
+    width: num_tiles_width,
+    height: num_tiles_height
+  }
+
+  snake = %{body: [{9, 9}], size: 5}
```

Comment:
`size: 5` is confusing here, I think we should use `size: 1`.

Reply:
👍 yes, that's just some things I quickly copy pasted over from my tryout code.

I already changed the snake to be more than one square large and with the respective "size" set 😉 

I could push another super rough WIP commit if you want. But I'd personally have waited until I have the wording for the section I'm currently writing done.

Reply:
No problem, push when ready.